profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ImageApiMockTest.java

```diff
@@ -0,0 +1,95 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.features;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import java.util.List;
+import org.apache.jclouds.profitbricks.rest.domain.Image;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksApiMockTest;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import org.testng.annotations.Test;
+
+@Test(groups = "unit", testName = "ImageApiMockTest", singleThreaded = true)
+public class ImageApiMockTest extends BaseProfitBricksApiMockTest {
+   
+   @Test
+   public void testGetList() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody(stringFromResource("/image/list.json"))
+      );
+      
+      List<Image> list = imageApi().getList();
+      
+      assertNotNull(list);
+      assertEquals(list.size(), 13);
+      assertEquals(list.get(0).properties().name(), "bacula-client-agent.iso");
+      
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/images");
+   }
+       
+   @Test
+   public void testGetImage() throws InterruptedException {
+      MockResponse response = new MockResponse();
+      response.setBody(stringFromResource("/image/get.json"));
+      response.setHeader("Content-Type", "application/vnd.profitbricks.resource+json");
+      
+      server.enqueue(response);
+      
+      Image image = imageApi().getImage("some-id");
+      
+      assertNotNull(image);
+      assertEquals(image.properties().name(), "CentOS-6.7-x86_64-netinstall.iso");
+      
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/images/some-id");
+   }
+     
+   @Test
+   public void testUpdate() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody(stringFromResource("/image/get.json"))
+      );
+      
+      api.imageApi().updateImage(
+              Image.Request.updatingBuilder()
+              .id("some-id")
+              .name("new-image-name")
+              .description("description...")
+              .build());
+            
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "PATCH", "/rest/images/some-id", "{\"name\": \"new-image-name\", \"description\": \"description...\"}");
+   }
+   
+   @Test
+   public void testDelete() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody("")
+      );
+      
+      imageApi().deleteImage("some-id");
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "DELETE", "/images/some-id");
+   }
+        
+   private ImageApi imageApi() {
+      return api.imageApi();
+   }
```

Comment:
> Rule of thumb is: if your method has a @Fallback annotation, it needs a test. Otherwise, fallback test is not needed.

+1.

So in this case: a 404/401 test won't be needed for `updateImage`, because it doesn't have a declared fallback (and jclouds will propagate an appropriate exception), but we need one for `getImage`, so that we can verify that the method's behavior indeed returns an empty list, instead of a `ResourceNotFoundException`/`AuthorizationException`.  

I got confused with this as well. Sorry for the confusion, @mirza-spc !